### ISSUE TITLE
Exhaustive case with enum accepts any predicate method

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#10383]
An exhaustive case with enum values accepts expressions to be constants (`Foo::BAR`) or predicate methods (`.bar?`). But it does not validate whether the predicate method actually represents an enum member. Any method is allowed.

```cr
enum Foo
  BAR

  def qux?
    true
  end
end

case Foo::BAR
in .qux? # QUX is not an enum member
  "qux"
in .bar?
  "bar"
end # => "qux"
```

This is, well, somewhat unexpected. But not necessarily a bug. I assume this just hasn't been considered.

This could be useful when you want to treat several members with the same predicate similarly. Technically you don't need to repeat the definition of this group of members. However, it doesn't impact exhaustiveness checks and the exhaustive case still requires expressions for all members. So it's not really of practical use. I think custom predicate methods only make sense with non-exhaustive case.

So I suggest this should be a compiler error.

**Comment by asterite** [comment]
Yeah, I think it should be a compiler error. I thought it would be doing that already.
